src/main/java/bio/terra/stairway/FlightDao.java

```diff
@@ -926,15 +926,16 @@ private void storeWorkingParameters(
         "SELECT key, value FROM "
             + FLIGHT_WORKING_TABLE
             + " WHERE flightlog_id="
-            + "(SELECT id FROM flightlog WHERE log_time="
-            + "   (SELECT MAX(log_time) FROM flightlog WHERE flightid = :flightId))";
+            + "(SELECT id FROM flightlog WHERE (flightid = :flightId) AND log_time="
+            + "   (SELECT MAX(log_time) FROM flightlog WHERE flightid = :flightId2))";
```

Comment:
Why does this need to be a new substitution variable?   : flightId and :flightId2 both get set to `flightId`

Reply:
I thought so too, but our [`NamedParameterPreparedStatement`](https://github.com/DataBiosphere/stairway/blob/develop/src/main/java/bio/terra/stairway/NamedParameterPreparedStatement.java) class will only substitute the first instance seen.  Fortunately unit tests caught it.

Probably worth considering enhancing `NamedParameterPreparedStatement` as a follow on, I can create a Jira ticket.  In the meantime this is a faster path to a fix and [there's precedent](https://github.com/DataBiosphere/stairway/blob/develop/src/main/java/bio/terra/stairway/FlightDao.java#L636).

Reply:
`NamedParameterPreparedStatement` is not a sophisticated implementation. I made it that way so that I didn't have to touch the values or deal with variable types.